### PR TITLE
Add package docs

### PR DESCRIPTION
**Pull request by MarcoPolo**
Adds some package docs that will show up in the godoc page. Inspired by the formatting of https://pkg.go.dev/encoding/gob.

### CODE REVIEW

**Comment by MarcoPolo** [comment]
Coming back to this, I think this makes more sense to live in the README. So I'm thinking I should merge this information into the existing information in the readme and add a table of contents at the top of the readme, and adjust the ordering so usage comes first.

Wdyt @marten-seemann ?

**Comment by BigLep** [comment]
@MarcoPolo : I know you weren't asking me, but your plan seems reasonable to me.  
I'll wait to review the text until we have the merged readme.
I assume go docs will link to the readme for more info so it's easily discoverable when someone's looking at the go docs?

**Comment by MarcoPolo** [comment]
updated @marten-seemann

File: README.md [hunk 1]
```diff
@@ -354,6 +274,126 @@ this would result in a limit of 1256 file descriptors for the system scope.
...
+
+Sometimes disabling all limits is useful when you want to see how much
+resources you use during normal operation. You can then use this information to
+define your initial limits. Disable the limits by using `InfiniteLimits`.
+
+### Debug "resource limit exceeded" errors
+
+These errors occur whenever we've hit a limit. For example we'll get this error
+if we are at our limit for the number of streams we can have, and we try to open
+one more.
```
Comment (marten-seemann) [inline README.md:310]: There's an inconsistency between "you" in the other paragraphs and the "we" here. Maybe it's better to use passive voice here: "These errors occur whenever a limit is hit."

File: README.md [hunk 2]
```diff
@@ -354,6 +274,126 @@ this would result in a limit of 1256 file descriptors for the system scope.
...
+
+By default the resource manager ships with some reasonable scaling limits and
+makes a reasonable guess at how much system memory you want to dedicate to the
+go-libp2p process. For the default definitions see `DefaultLimits` and
+`ScalingLimitConfig.AutoScale()`.
+
+### Tweaking Defaults
+
+If the defaults seem mostly okay, but you want to adjust one facet you can do
+simply copy the defaults and update the field you want to change. You can
```
[the remark this answers is not in the document]
Comment (MarcoPolo) [reply README.md:287]: Maybe an example would help?

File: README.md (hunk 1 above)
Comment (MarcoPolo) [reply README.md:310]: Used the passive voice for the first sentence, then `you` for the rest of the paragraph.

**Pull request merged**